app/services/onestop_id.rb

```diff
@@ -87,6 +87,15 @@ class StopOnestopId < OnestopIdBase
   class RouteOnestopId < OnestopIdBase
     PREFIX = :r
     MODEL = Route
+
+    def to_s
+      geohash = @geohash[0...self.class::GEOHASH_MAX_LENGTH]
+      [
+        self.class::PREFIX,
+        geohash,
+        @name[0...RouteStopPatternOnestopId.max_name_length(geohash.length)],
+      ].join(COMPONENT_SEPARATOR)[0...self.class::MAX_LENGTH]
+    end
```

Comment:
Write comment to mention we're reserving 14 characters (2x6 + 2 separators) for RSP components.